docs/usage.md

```diff
@@ -0,0 +1,86 @@
+# Usage
+AzureAuth is a generic Azure Active Directory credential provider. It currently supports the following modes of [public client authentication](https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-client-applications) (i.e., authenticating a human user.)
+* WAM (Windows Only)
+* Embedded Web View (Windows Only)
+* System Web Browser (Used on OSX in-place of Embedded Web View)
+* Device Code Flow (no GUI, terminal interface only).
+
+## Requirements
+This CLI is a "pass-through" for using [MSAL.NET](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet). This means it does not provide any client ID (aka app registration) by default. You must register and configure your own app registration to authenticate with.
+
+### Configure your App Registration
+1. You can follow [this quick start guide](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app) to setup your application.
+In order to support public client auth modes you must also add redirect URIs under "Mobile and Desktop applications" in the Authentication blade of your app registration in the Azure Portal.
+2. To support WAM (the Windows broker), you must add
+   ```
+   ms-appx-web://microsoft.aad.brokerplugin/<CLIENT_ID>
+   ```
+3. To support system web browser, you must add
+   ```
+   http://localhost
+   ```
+   (Note - do not use `https` here, this is for local redirect and TLS won't work here.)
+4. In the bottom of the Authentication Blade, enable the "Allow public client flows" setting.
```

Comment:
Do you mean you think it is currently better documented by the Azure docs or that it would be better if the Azure docs did cover this? Because they do not and it is not clear or easy to figure out.

Reply:
And this is required to have done to make use of `azureauth`

Reply:
Since we need this information, I'm not changing the content.